### PR TITLE
Support for prims collapse op (lowering to linalg)

### PR DESCRIPTION
**Pull request by newling**
Steps taken:
1) add generator code to torch_ods_gen.py, run update_torch_ods.sh
2) add (custom) shape and type inference generator code to abstract_interp_lib_gen.py, run update_abstract_interp_lib.sh
3) Implement lowering to tensor.collapse_dims. Requires the `start` and `end` values to be constant, else lowering fails
4) Update xfail_sets.py (append to LTC_XFAIL_SET) after running /tools/e2e_test.sh --filter Collapse --verbose -c XX for all support backends (XX). 

Motivation: 
- Supporting the collapse operation will be useful for lowering of pixel_shuffle (see Issue #2559)

### CODE REVIEW

File: projects/pt1/python/torch_mlir_e2e_test/test_suite/reshape_like.py [hunk 1]
```diff
@@ -122,6 +122,48 @@ def forward(self, a):
...
+
+
+@register_test_case(
+    module_factory=lambda: CollapseStaticModule())
+def CollapseStaticModule_basic(module, tu: TestUtils):
+    module.forward(tu.rand(2,3,4))
+
+# ==============================================================================
+#
+class CollapseDynamicModule(torch.nn.Module):
```
Comment (ramiro050) [inline projects/pt1/python/torch_mlir_e2e_test/test_suite/reshape_like.py:147]: I imagine the lowering works for fully dynamic inputs too, right? If so, we should have a test case that is fully dynamic (this one collapses static with dynamic). Also, can we test the case where the collapse range encompasses more than two elements?

**Review by ramiro050 — CHANGES_REQUESTED** [review]
Thanks!

File: projects/pt1/python/torch_mlir_e2e_test/test_suite/reshape_like.py (hunk 1 above)
Comment (newling) [reply projects/pt1/python/torch_mlir_e2e_test/test_suite/reshape_like.py]: Added these 2 tests. I also added a test where the input is rank-1, which revealed a bug (basically upstream MLIR doesn't allow collapse_shape if nothing collapses) Always a good idea asking for extra tests :-D

**Review by ramiro050 — APPROVED** [review]

File: projects/pt1/python/torch_mlir_e2e_test/test_suite/reshape_like.py (hunk 1 above)
Comment (ramiro050) [reply projects/pt1/python/torch_mlir_e2e_test/test_suite/reshape_like.py]: Nice!

**Pull request merged**